controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -420,20 +422,56 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
       });
       return handleError<PlainMessage<GetFederatedSubgraphSDLByNameResponse>>(logger, async () => {
         const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
+        const fedRepo = new FederatedGraphRepository(opts.db, authContext.organizationId);
         const subgraphRepo = new SubgraphRepository(opts.db, authContext.organizationId);
+        const compositionRepo = new GraphCompositionRepository(opts.db);
         const subgraph = await subgraphRepo.byName(req.name);
-        if (subgraph?.schemaSDL) {
+        if (!subgraph) {
+          return {
+            response: {
+              code: EnumStatusCode.ERR_NOT_FOUND,
+            },
+          };
+        }
+
+        const fedGraphSchemaVersion = await fedRepo.getLatestValidFederatedGraphSchemaVersion(req.federatedGraphName);
```

Comment:
Should we rename it to `getLatestValidSchemaVersion`?

Reply:
Done